frame/clearing-house/src/lib.rs

```diff
@@ -305,19 +368,78 @@ pub mod pallet {
 			amount: T::Balance,
 		) -> DispatchResult {
 			let acc = ensure_signed(origin)?;
-			<Self as MarginTrading>::add_margin(&acc, asset, amount)?;
+			<Self as ClearingHouse>::add_margin(&acc, asset, amount)?;
+			Ok(())
+		}
+
+		/// # Overview
+		/// Creates a new perpetuals market with the desired parameters.
+		///
+		/// ## Parameters
+		/// - `asset`: Asset id of the underlying for the derivatives market
+		/// - `vamm_params`: Parameters for creating and initializing the vAMM for price discovery
+		/// - `margin_ratio_initial`: Minimum margin ratio for opening a new position
+		/// - `margin_ratio_maintenance`: Margin ratio below which liquidations can occur
```

Comment:
Shouldn't `origin`, `funding_frequency` and `funding_period` be added to the list?

Reply:
Extrinsics always have an origin, so I think that'd be redundant. The other two are missing indeed